resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -60,9 +59,9 @@
         this.nameServerAddr = nameServerAddr;
         this.question = question;
         this.promise = promise;
-
-        id = allocateId();
         recursionDesired = parent.isRecursionDesired();
+        id = parent.queryContextManager.add(this);
```

Comment:
I guess that's not gonna happen because `DnsQueryContext` is only accessed later when a matching DNS response is received.